### DISCUSSION TITLE
Error while uploading an image using Minio plugin S3 API Requests must be made to API port. InvalidArgument: S3 API Requests must be made to API port.

### DISCUSSION THREAD

**Discussion by Gunkev** [medusajs/medusa#3542] [Q&A]
Hello I am getting this error below while trying to add a product in the backend medusa dashboard
![minio uplad error](https://user-images.githubusercontent.com/36812576/226661581-2e1ad37c-a1f4-440b-811b-1323d085dbb4.PNG)

I am using windows 10
Nextjs storefront
I installed Minio. command to run it:  .\minio.exe server E:\minio --address :9001 --console-address :9090
I installed the Minio plugin in my project

Minio backend/.env variables:
```
MINIO_ENDPOINT=http://127.0.0.1:9090
MINIO_BUCKET=electronics
MINIO_ACCESS_KEY=oooooooooooooooo
MINIO_SECRET_KEY=oooooooooooooooooooooo
```

**NB: the o are not my actual keys**

backend/medusa-config.js

```
const plugins = [
  `medusa-fulfillment-manual`,
  `medusa-payment-manual`,
  {
    resolve: `medusa-file-minio`,
    options: {
      endpoint: process.env.MINIO_ENDPOINT,
      bucket: process.env.MINIO_BUCKET,
      access_key_id: process.env.MINIO_ACCESS_KEY,
      secret_access_key: process.env.MINIO_SECRET_KEY,
    },
  },
];
```

storefront/next.config.js

```
module.exports = withStoreConfig({
  features: store.features,
  reactStrictMode: true,
  images: {
    domains: [
      "medusa-public-images.s3.eu-west-1.amazonaws.com",
      "localhost",
      "127.0.0.1",
    ],
  },
})
```

**Comment by FadySabry712** [comment]
did you get any help??

**Reply by KeaZulu23** [reply]
i also need help

**Comment by genox** [comment]
Some suggested running medusa under WSL as it is not tested on windows.

I run minio and medusa on Debian and macOS and everything seems to work fine here.

**Comment by nelsonokothguya** [comment]
Looking for a solution to this. Any suggestions will be appreciated.

**Comment by kossumov** [comment]
Port 9090 is the web console port
Port 9000 is the S3 API port
Use 9000

**Comment by k1enttt** [comment]
Try to change MINIO_ENDPOINT in .env to http://127.0.0.1:9001. It works for me
